wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +37,380 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
     LockedWallet(nodeApi, chainQueryApi)
   }
 
+  /** We need a output so we can get proper tx size estimation */
+  private def getDummyDLCFundingOutputs(
+      amount: CurrencyUnit): Vector[TransactionOutput] = {
+    val key = ECPublicKey.freshPublicKey
+    val spk = P2WSHWitnessSPKV0(MultiSignatureScriptPubKey(2, Seq(key, key)))
+
+    // funded output and the other party's change
+    Vector(TransactionOutput(amount, spk), TransactionOutput(Satoshis.one, spk))
+  }
+
+  override def createDLCOffer(
+      amount: Bitcoins,
+      oracleInfo: OracleInfo,
+      contractInfoVec: Vector[Sha256DigestBE],
+      feeRateOpt: Option[SatoshisPerVirtualByte],
+      locktime: UInt32,
+      refundLocktime: UInt32): Future[DLCOffer] = {
+    logger.debug("Calculating relevant wallet data for DLC Offer")
+
+    val contractInfo =
+      ContractInfo(
+        (Map.newBuilder ++= contractInfoVec.map(sha => (sha, amount.satoshis)))
+          .result())
+
+    val timeouts = DLCTimeouts(5,
+                               BlockStamp(locktime.toInt),
+                               BlockStamp(refundLocktime.toInt))
+
+    val feeRate = feeRateOpt.getOrElse(SatoshisPerVirtualByte.one)
```

Comment:
ditto above comment about optional fees. The only time I think this is okay is if the wallet is doing actual feerate estimation (as opposed to defaulting to `1`)

Reply:
so my opinion on this is we should make a `getFeeRate()` methind in our wallet and just return a default value for now, so we don't have a bunch of `feeRateOpt.getOrElse(Sats.one)` laying around. Then at least we can do `feeRateOpt.getOrElse(wallet.getFeeRate)`